theano/sandbox/cuda/dnn_gw.c

```diff
@@ -30,6 +25,21 @@ APPLY_SPECIFIC(conv_gw)(CudaNdarray *input, CudaNdarray *output,
     return 1;
 #endif
 
+
+  if (CudaNdarray_HOST_DIMS(input)[0] == 0) {
+    // zero batch size. Return a zero gradient without calling cuDNN (which will fail on zero batch size)
+    Py_XDECREF(*kerns);
+    *kerns = (CudaNdarray *) CudaNdarray_ZEROS(CudaNdarray_NDIM((*kerns)), (int *) CudaNdarray_HOST_DIMS((*kerns)));
+    if (*kerns == NULL) return 1;
+    Py_INCREF(*kerns);
```

Comment:
I wasn't sure whether I need the Py_INCREF here -- assumed I did given similar INCREF above on line 20, but wanted to check.

I also wasn't sure if it's OK to return a new zeros array in the case where CONV_INPLACE is specified, hopefully it is as it looked quite a pain in general to zero out an existing array in-place.